### DISCUSSION TITLE
How to use the Rig Control function?

### DISCUSSION THREAD

**Discussion by n2iw** [AlexandreRouma/SDRPlusPlus#380] [Q&A] [answered]
I'm really happy to report that I'm running SDR++ on a M1(Apple silicon/ARM) Mac mini with SDRPlay RSP1A. I used MacPorts to install all the dependencies, including SDRPlay API. It's all under Rosetta instead of native ARM, because SDRPlay API is not open source and  binary release currently only support Intel chips. I've created a ticket with SDRPlay, hopefully they will fix it soon.

My goal is to have a working bandscope/pandadapter, meaning when I changed frequency on my radio, I want SDR++ to follow my radio. The key here is the rig control module in SDR++, I couldn't find much information about it. What protocol it supports, what radio it emulates etc. Any information would be appreciated, thanks!

**Comment by AlexandreRouma** [comment, marked as answer]
Hello,
SDR++ doesn't emulate a particular radio, it just implement the basic RigCTL server calls. I don't think this would allow to control a rig directly as that requires a RigCTL client which SDR++ doesn't have yet. You might be able to write a script that keeps polling SDR++ for its status and then sends it to the rig.

**Reply by n2iw** [reply]
Thanks for the super quick reply! Actually, I'm not trying to control my radio from SDR++ (yet), what I'm trying to do is to control SDR++ from my logging software RUMlogNG. I was not aware of RigCTL, quick google indicates its related to Hamlib. I'll dig a little deeper and possibly work with RUMlogNG's author to add support for Hamlib.

Thanks again for the quick reply!

**Reply by AlexandreRouma** [reply]
In this case, you just need to select "hamlib NET rigctl" (or similar, that's what WSJT-X calls the option) and enter the IP and port (by default 127.0.0.1:4532 by check the rigctl server menu) and it should, connect to sdr++

**Reply by n2iw** [reply]
Thanks! My log software RUMlog doesn't support Hamlib yet. I just requested the author to add it. Figures crossed!

**Reply by snwvo7** [reply]
Gents,
I have a similar issue to receive via Rigctl Server.
I am using Macdoppler for doppler corrections via satelites and want to use SDR++ with SDRplay as an doppler corrected SDR receiver.
I have configured Macdoppler to send information over UDP via ip:192.168.2.5 Port: 9932.
A UDP listner is installed to check for data output of Macdoppler, that's oke.
I have configured Rigctl Server in SDR++  (192.168.2.5 Port 9932).
SDR++ is listening but no frequency updates and no  error messages.
Any ideas how to solve this?
Thanks, Martin PD1MV

**Reply by AlexandreRouma** [reply]
Rigctl is not UDP. No idea what protocol that software is using but it sure isn't anything supported.
Use a more modern satellite tracker like gpredict instead.

**Reply by snwvo7** [reply]
Thanks for the quick response Alexsandre, Macdoppler is the best satellite tracker for MacOS so far.
It is a bit of a struckle to let apps communicate with each other  on MAC. I would have been better when they all use Hamlib.

**Reply by jstromsoe** [reply]
Does SDR++ support rigctrl client?  For anyone using it with a radio this seems like one of the absolute most basic functions possible.  All it needs to do is send/receive commands to a prescribed serial device and support the protocol.

**Reply by AlexandreRouma** [reply]
https://www.sdrpp.org/manual.pdf page 7
> Rigctl client - allows use of SDR++ as panadapter (not yet in nightly releases but may
be built)
